src/main/java/com/google/devtools/build/lib/sandbox/LinuxSandboxedSpawnRunner.java

```diff
@@ -178,6 +185,23 @@ protected SandboxedSpawn prepareSpawn(Spawn spawn, SpawnExecutionContext context
     sandboxExecRoot.getParentDirectory().createDirectory();
     sandboxExecRoot.createDirectory();
 
+    PathFragment tmpRoot = PathFragment.create("/tmp");
+    Optional<PathFragment> tmpfsPathUnderTmp = getSandboxOptions().sandboxTmpfsPath.stream()
+      .filter(path -> path.startsWith(tmpRoot))
+      .findFirst();
+    // Mounting a tmpfs strictly below the hermetic /tmp isn't supported. Mounting a tmpfs on /tmp
+    // makes mounting the disk-based hermetic /tmp unnecessary.
+    if (tmpfsPathUnderTmp.isPresent() && !getSandboxOptions().sandboxTmpfsPath.contains(tmpRoot)
```

Comment:
The ifs for the warning and the actual logic should not be separate. It would be better to have one if with the logic and an inner if with just the `warnedAboutNonHermeticTmp` check.

Reply:
I rearranged the control flow and also moved the comments around. PTAL.